tsdb/wlog/live_reader.go

```diff
@@ -195,39 +193,28 @@ func (r *LiveReader) buildRecord() (bool, error) {
 
 		rt := recTypeFromHeader(r.hdr[0])
 		if rt == recFirst || rt == recFull {
-			r.rec = r.rec[:0]
-			r.compressBuf = r.compressBuf[:0]
+			r.precomprBuf = r.precomprBuf[:0]
 		}
 
-		isSnappyCompressed := r.hdr[0]&snappyMask == snappyMask
-		isZstdCompressed := r.hdr[0]&zstdMask == zstdMask
-
-		if isSnappyCompressed || isZstdCompressed {
-			r.compressBuf = append(r.compressBuf, temp...)
-		} else {
-			r.rec = append(r.rec, temp...)
+		// TODO(bwplotka): Handle unknown compressions.
```

Comment:
just a very naive comment, could we just `return false, newError` if we find an unknown compression here? Is that ok or do we want to gracefully handle it?

Reply:
No, good question. We can't. We would need to change WAL format as we have only 3 bits for exactly those 3 options AFAIK, so by "handling unknown compressions" I mean changing protocol one day. I will add more details to this comment.

Reply:
Removing todo and adding proper comment about this.